python/paddle/fluid/contrib/slim/core/compressor.py

```diff
@@ -188,7 +189,8 @@ def run_eval_graph(self, sampled_rate=None, cached_id=0):
             reader = cached_reader(reader, sampled_rate, self.cache_path,
                                    cached_id)
 
-        if isinstance(reader, Variable):
+        if isinstance(context.train_reader, Variable) or isinstance(
```

Comment:
Why change eval_reader to context.train_reader?

Reply:
Fixed.